### PR TITLE
Add file type on server side

### LINKED ISSUE #80: integrate mime\type module to check for file types
###### FEATURE
---
In order to open photos, audio files, video files, pdfs, etc. safely on the client.

https://www.npmjs.com/package/mime-types

### LINKED ISSUE #131: settingsService prints out the wrong assertion error on NO MATCH FOUND

### PR DESCRIPTION
**Pull request by Thatkookooguy**
# Change Summary

 - Passing to client the mime-type of each file [resolves #80]
 
 > Also, did the changes needed to support this on the client-side. This means ***changing syntax by mime-type***
 
 - fixed some issues with settingsService + some nits :-) [resolves #131]
 
- [x] did you link this PR to an issue?
- [x] did you lint your changes to both javascript and scss?
- [x] "I'm pretty sure I'll be able to read and understand this PR, even if I wasn't the author." - ***said the PR author***

### CODE REVIEW

**Comment by Thatkookooguy** [comment]
Fixed a closing tag issue in the master branch. noticed it since kibibit changed the syntax of `home.html` to `html` when I opened it in the demo, and `ace.js` showed me a warning in the gutter :-)

File: public/app/components/codeEditor/codeEditor.js [hunk 1]
```diff
@@ -73,43 +76,18 @@ angular.module('kibibitCodeEditor')
...
+        FileService.getFile(filePath, function(fileInfo) {
+          vm.fileInfo = fileInfo.data;
+          vm.code = vm.fileInfo.content;
         });
       }
     };
-  }
-])
-
-.directive('kbChangeAceScroll', function() {
```
Comment (ortichon) [inline public/app/components/codeEditor/codeEditor.js]: @Thatkookooguy 
What about this directive?

**Comment by ortichon** [comment]
@Thatkookooguy 
I know we want to show `svg` files using the image-editor, but will we able to view `svg` in "code" mode?

**Comment by ortichon** [comment]
@Thatkookooguy 
Anyway, it looks great. :)

**Comment by Thatkookooguy** [comment]
@ortichon I made it invalid for now until you actually see a compiled SVG view.

Because you can't actually see what you're doing :-)

File: public/app/components/codeEditor/codeEditor.js (hunk 1 above)
Comment (Thatkookooguy) [inline public/app/components/codeEditor/codeEditor.js]: It wasn't attached anywhere. It was a leftover from an experiment I did in a previous PR

**Comment by ortichon** [comment]
LGTM

**Pull request merged**